src/utils/data-access.js

```diff
@@ -69,7 +69,7 @@ export async function syncSuggestions({
   await Promise.all(
     existingSuggestions
       .filter((existing) => {
-        const existingKey = buildKey(existing);
+        const existingKey = buildKey(existing.data);
```

Comment:
@dzehnder my unit tests caught a bug here. If I understand this part correctly, the `data` object is under the existing `suggestion` object. Therefore we need to `buildKey(existing.data)` instead of `buildKey(existing)`

can you confirm this please?

Reply:
Do we need this fix in above block as well at Line 64?

Also, Is this filter redundant, since we are already doing "// Remove outdated suggestions" just above this?

Reply:
this is fixed by @dzehnder and I updated this branch ✅